modules/deriving/src/main/scala/shapeless3/deriving/kinds.scala

```diff
@@ -150,10 +150,18 @@ object K0:
       inst.asInstanceOf
 
   extension [F[_], T](inst: CoproductInstances[F, T])
+    inline def instanceIndex(t: T): Int =
+      inst.mirror.ordinal(t.asInstanceOf)
```

Comment:
The signature here is wrong `T` is the type class, not the enum so we can't get an ordinal from the mirror. Also what is we just exposed `mirror.ordinal` in the erased instances?